File: typescript/src/models/checkPerformedType.ts
// this file is @generated

export enum CheckPerformedType {
  Avs = "avs",
  Cvv = "cvv",
  Ani = "ani",
  Auth0 = "auth0",
}

export const CheckPerformedTypeSerializer = {
  _fromJsonObject(object: any): CheckPerformedType {
    return object;
  },

  _toJsonObject(self: CheckPerformedType): any {
    return self;
  },
};

File: rust/src/models/check_performed_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum CheckPerformedType {
    #[serde(rename = "avs")]
    Avs,
    #[serde(rename = "cvv")]
    Cvv,
    #[serde(rename = "ani")]
    Ani,
    #[serde(rename = "auth0")]
    Auth0,
}

impl fmt::Display for CheckPerformedType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::Avs => "avs",
            Self::Cvv => "cvv",
            Self::Ani => "ani",
            Self::Auth0 => "auth0",
        };
        f.write_str(value)
    }
}
